docs/concepts/gnovm-evm-cosmwasm.md

```diff
@@ -0,0 +1,162 @@
+
+Meanwhile, the GnoVM operates entirely on Gno, which is [99% identical to Go](https://gno.land/r/gnoland/blog:p/intro), a complete and widely used programming language. Developers new to blockchain are much more likely to know Go than they are to know Solidity or Rust. The ability for developers to tap into the Go ecosystem and access type checking, garbage collection, and many of the standard libraries, among other features and tooling, makes Gno highly usable even compared to Rust—and certainly compared to a narrowly specialized language like Solidity.
+
+## State Persistence Boosts Efficiency
+
+Smart contracts on the GnoVM have state persistence, meaning they can be frozen and resumed and their memory will be automatically saved. This decreases the likelihood of programming errors and provides a smoother development experience. It also improves program succinctness, because programmers need not manually reference databases to store objects in memory.
+
+CosmWasm's state management is made more complex by its use of actor-model message passing. This complexity places a higher burden on the developer, who needs to deal with manual state management across multiple files for a single contract. The EVM is not much better, also requiring developers to handle state explicitly using serialization.
+
+## The Go Ecosystem Provides Security
+
+The EVM is known to be [vulnerable to reentrancy attacks](https://docs.cosmwasm.com/docs/architecture/smart-contracts/), which occur when a function externally calls an untrusted contract. Both the GnoVM and CosmWasm offer robust protection against this type of attack. CosmWasm entirely prevents any contract from directly calling another contract, while the GnoVM's design restricts direct memory access and handles state management internally.
+
+The GnoVM also leverages the strong security design of Go. The language's simplicity reduces the chance of programming errors, as does its reliance on the principle of least privilege. Gno developers can also tap into the robust third-party tooling of the Go ecosystem. For example, they can access tools for static analysis that would make it easier to design secure smart contracts.
+
+## On-chain Readability Promotes Transparency
+
+Lowering the barrier to entry for web3 smart contract solutions is not just about developer onboarding–it’s also about the accessibility of information. To this point, one of the most crucial differences between the GnoVM, the EVM and CosmWasm lies in the verifiability of their smart contracts. While the GnoVM interpreter parses Gno code into an abstract syntax tree before executing it, the EVM and CosmWasm must first convert Solidity and Rust into bytecode. This means that GnoVM smart contracts are recorded on the blockchain in human-readable code, while the EVM and CosmWasm blockchains only have machine-readable bytecode. The transparency is unique among blockchains.
```

Comment:
I think the way these blockchains store source code is not a "key difference", but just a design choice. For example, a better highlight would've been that you cannot import other smart contracts easily in Solidity, or have type safety for external types without you yourself uploading the contract that contains them.

We can just as easily switch to a bytecode approach tomorrow with Gno (something we are actively thinking about), and then this difference you've mentioned wouldn't matter at all

Reply:
Thank you for all of these helpful comments. I’ve just committed changes that address your other edits, but I’m blocked on this one. Could you please clarify the following for me?
1. What aspects of Solidity make importing other smart contracts more difficult than in the GnoVM? Where does CosmWasm rest on this spectrum?
2. I’m having trouble understanding how CosmWasm deals with type safety for external types. Could you clarify? 
3. Would you like me to eliminate all the discussion of the GnoVM blockchain storing AST vs. the others storing bytecode? I included it because it’s discussed in the Gno.land intro blog and Jae Kwon brought it up in his DEVMOS talk, but I can cut it completely that is preferable.